tests/lib/openstack_query/query_blocks/test_query_parser.py

```diff
@@ -1,468 +1,466 @@
-        self.assertEqual(self.instance._group_by, MockProperties.PROP_1)
-        mock_parse_group_ranges.assert_called_once_with(mock_group_ranges)
-
-    @patch("openstack_query.query_blocks.query_parser.QueryParser._parse_group_ranges")
-    @patch(
-        "openstack_query.query_blocks.query_parser.QueryParser._add_include_missing_group"
-    )
-    def test_parse_group_by_with_group_ranges_and_include_missing(
-        self, mock_add_include_missing_group, mock_parse_group_ranges
-    ):
-        """
-        Tests parse_group_by functions expectedly - with group ranges
-        Should set group_by and call parse_group_ranges
-        """
-        mock_group_ranges = {"group1": ["val1", "val2", "val3"], "group2": ["val4"]}
-        self.instance.parse_group_by(
-            MockProperties.PROP_1, mock_group_ranges, include_missing=True
-        )
-        self.assertEqual(self.instance._group_by, MockProperties.PROP_1)
-        mock_parse_group_ranges.assert_called_once_with(mock_group_ranges)
-        mock_add_include_missing_group.assert_called_once_with(mock_group_ranges)
-
-    def test_run_parse_group_ranges(self):
-        """
-        Tests parse_group_ranges functions expectedly
-        Should setup groups based on given group mappings
-        """
-        mock_group_by = MockProperties.PROP_1
-        self.instance._group_by = mock_group_by
-        mock_group_ranges = {"group1": ["val1", "val2", "val3"], "group2": ["val4"]}
 
-        with patch.object(
-            MockProperties, "get_prop_mapping", wraps=self._get_prop_func
-        ) as mock_get_prop_func:
-            self.instance._parse_group_ranges(group_ranges=mock_group_ranges)
-            mock_get_prop_func.assert_called_once_with(mock_group_by)
+@pytest.fixture(name="instance")
+def instance_fixture():
+    """
+    Returns an instance with mocked prop_enum_cls inject
+    """
+    mock_prop_enum_cls = MockProperties
+    return QueryParser(prop_enum_cls=mock_prop_enum_cls)
 
-        self.assertTrue(self.instance._group_mappings["group1"]({"prop_1": "val1"}))
-        self.assertFalse(self.instance._group_mappings["group2"]({"prop_1": "val1"}))
 
-    def test_add_include_missing_group(self):
-        mock_group_by = MockProperties.PROP_1
-        self.instance._group_by = mock_group_by
-        mock_group_ranges = {"group1": ["val1", "val2", "val3"], "group2": ["val4"]}
+@pytest.fixture(name="mock_get_prop_mapping")
+def get_prop_mapping_fixture():
+    """
+    Returns a mocked get_prop_mapping function
+    """
 
+    def _mock_get_prop_mapping(mock_prop: MockProperties):
+        return MagicMock(wraps=lambda obj, prop=mock_prop: obj[prop.name.lower()])
+
+    return _mock_get_prop_mapping
+
+
+@pytest.fixture(name="run_sort_by_tests")
+def run_sort_by_tests_fixture(instance, mock_get_prop_mapping):
+    def _run_sort_by_tests(obj_list, sort_by_specs, expected_list):
+        instance.sort_by = sort_by_specs
         with patch.object(
```

Comment:
Could we add a docstring to this method?

Why is the method within the `run_sort_by_tests_fixture` method?

Reply:
This fixture is basically setup to return some shared boilerplate code to run sort by test cases - so I return a function which each test can run internally. 

I think its cleaner since if we want any setup code in the fixture (like here this fixture is using another `instance`) fixture - this can be separated from the actual test cases. In other words - the stuff in the internal function is stuff concerned with running the test case, the stuff outside the internal function is around setting up tests